### PR TITLE
Alling ND-ARP-Proxy with the Nokia suggestions

### PR DESCRIPTION
**Pull request by sbarguil**

### CODE REVIEW

File: yang/ietf-l2vpn-ntw.yang [hunk 1]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
-                          description
-                            "Enables (when set to 'true') or disables (when set
-                             to 'false') ARP suppression.";
-                        }
-                        leaf nd-proxy {
-                          if-feature "vpn-common:ipv6";
-                          type boolean;
-                          default "false";
+                        container arp-proxy {
+                          leaf arp-proxy-enable {
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1049]:
```suggestion
                          leaf enable {
```

File: yang/ietf-l2vpn-ntw.yang [hunk 2]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
-                          default "false";
-                          description
-                            "Enables (when set to 'true') or disables (when set
-                             to 'false') ARP suppression.";
-                        }
-                        leaf nd-proxy {
-                          if-feature "vpn-common:ipv6";
-                          type boolean;
-                          default "false";
+                        container arp-proxy {
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1048]:
```suggestion
                        container arp-proxy {
                          if-feature "vpn-common:ipv4";
```

File: yang/ietf-l2vpn-ntw.yang [hunk 3]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
+                          leaf arp-proxy-enable {
+                            type boolean;
+                            default "false";
+                            description
+                              "Enables (when set to 'true') or disables (when set
+                               to 'false') ARP proxy.";
+                          }
+                          leaf arp-suppression {
+                            if-feature "vpn-common:ipv4";
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1057]:
```suggestion
```

File: yang/ietf-l2vpn-ntw.yang [hunk 4]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
+                            type boolean;
+                            default "false";
+                            description
+                              "Enables (when set to 'true') or disables (when set
+                               to 'false') ARP suppression.";
+                            reference
+                              "draft-ietf-bess-evpn-proxy-arp-nd-13
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1064]:
```suggestion
                              "```

Comment (boucadair) [reply yang/ietf-l2vpn-ntw.yang:1064]: We can't cite I-Ds. If there is text from that I-d to discuss what arp suppression is about, we can include it in the description

File: yang/ietf-l2vpn-ntw.yang [hunk 5]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
+                              RFC 7432: BGP MPLS-Based Ethernet VPN";
+                          }
+                          leaf ip-mobility-threshold {
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1067]: Where is this defined?

File: yang/ietf-l2vpn-ntw.yang [hunk 6]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
                           description
-                            "Enables (when set to 'true') or disables (when set
-                             to 'false') ND proxy.";
+                            "Top container for the ARP Proxy";
                         }
-                        leaf nd-suppression {
-                          if-feature "vpn-common:ipv6";
-                          type boolean;
-                          default "false";
+                        container nd-proxy {
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1088]:
```suggestion
                        container nd-proxy {
                          if-feature "vpn-common:ipv6"; 
```

File: yang/ietf-l2vpn-ntw.yang [hunk 7]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
+                          leaf nd-proxy-enable {
+                            if-feature "vpn-common:ipv6";
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1090]:
```suggestion

```

File: yang/ietf-l2vpn-ntw.yang [hunk 8]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
-                            "Enables (when set to 'true') or disables (when set
-                             to 'false') ND proxy.";
+                            "Top container for the ARP Proxy";
                         }
-                        leaf nd-suppression {
-                          if-feature "vpn-common:ipv6";
-                          type boolean;
-                          default "false";
+                        container nd-proxy {
+                          leaf nd-proxy-enable {
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1089]:
```suggestion
                          leaf enable {
```

File: yang/ietf-l2vpn-ntw.yang [hunk 9]
```diff
@@ -1045,36 +1045,87 @@ module ietf-l2vpn-ntw {
...
+                            description
+                              "Enables (when set to 'true') or disables (when set to 'false') 
+                              ND proxy. If true the NDP queries for an IP address that 
+                              is not on that network are suppressed. NDP suppression is a technique 
+                              used to reduce the amount of NDP Packets flooding within individual 
+                              VXLAN segments, that is between VMs connected to the same logical 
+                              switch.";
+                          }
+                          leaf nd-suppression {
+                            if-feature "vpn-common:ipv6";
```
Comment (boucadair) [inline yang/ietf-l2vpn-ntw.yang:1102]:
```suggestion

```

**Review by boucadair — CHANGES_REQUESTED** [review]

**Comment by boucadair** [comment]
@sbarguil: Any udpate about the requested changes and clarifications? Thank you.

File: yang/ietf-l2vpn-ntw.yang (hunk 5 above)
Comment (sbarguil) [reply yang/ietf-l2vpn-ntw.yang:1067]: Sorry the delay. It Is defined https://datatracker.ietf.org/doc/html/draft-ietf-bess-evpn-proxy-arp-nd-14#section-3.7

**Pull request merged**